### ISSUE TITLE
Had Trouble when Reproduced TrOCR-Large-IAM

### ISSUE THREAD

**Issue by XHMY** [microsoft/unilm#831]
I am using **TrOCR**. I want to reproduce the score of CER on the IAM dataset. But I only got `Generate test with beam=10: CER2: 4.29`, which should be 2.89 as mentioned  in the [README.md](https://github.com/microsoft/unilm/tree/master/trocr#readme)

I trained on 8 RTX 3090 GPUs. I used the training command of [Fine-tuning on IAM](https://github.com/microsoft/unilm/tree/master/trocr#fine-tuning-on-iam). My complete training command is:

```shell
export BSZ=8
export valid_BSZ=16

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python -m torch.distributed.launch --nproc_per_node=8 \
$(which fairseq-train) \
--data-type STR --user-dir ./ --task text_recognition \
--arch trocr_large \
--seed 1111 --optimizer adam --lr 2e-05 --lr-scheduler inverse_sqrt \
--warmup-init-lr 1e-8 --warmup-updates 500 --weight-decay 0.0001 --log-format tqdm \
--log-interval 10 --batch-size ${BSZ} --batch-size-valid ${valid_BSZ} --save-dir ${SAVE_PATH} \
--tensorboard-logdir ${LOG_DIR} --max-epoch 300 --patience 20 --ddp-backend legacy_ddp \
--num-workers 8 --preprocess DA2 --update-freq 1 \
--bpe gpt2 --decoder-pretrained roberta \
--finetune-from-model trocr-large-stage1.pt --fp16 \
--keep-best-checkpoints 3 --keep-last-epochs 2  --save-interval 1 \
${DATA}
```
The training outputs: [train.log](https://github.com/microsoft/unilm/files/9386181/train.log)
The Tensorboard log: [log_ft_iam_large.tar.gz](https://github.com/microsoft/unilm/files/9386179/log_ft_iam_large.tar.gz)

My environments as follows:
- torch `1.8.2`
- torchaudio `0.8.2`
- torchvision `0.9.2`
- fairseq `0.12.2`
- CUDA `11.1.105`
- apex `0.1`

Could you help me to find out the problem?

**Comment by liminghao1630** [comment]
Hi, @XHMY, what training set do you use?

**Comment by XHMY** [comment]
> Hi, @XHMY, what training set do you use?

I followed the [Step-by-step Training Guide Using IAM Database](https://github.com/jpuigcerver/Laia/tree/master/egs/iam), and get the IAM training set. The screenshot of the training set is as follows.
<img width="1544" alt="WX20220822-144846@2x" src="https://user-images.githubusercontent.com/28671304/185857326-3b6f75aa-2c92-4efd-9d6a-630f1eb3c4a5.png">

**Comment by liminghao1630** [comment]
We take the training set and the validation set given in Laia as a whole as the new training set, and you can use the "General Dataset" class to implement this merge.

**Comment by XHMY** [comment]
May I merge the training set and validation set use a text editor, which simply append the `gt_valid.txt` to the `gt_train.txt`. I tried this way but still does not look fine. The CER2 was `4.27` after merging.

**Comment by liminghao1630** [comment]
@XHMY This way is ok I think. You may need to check the other aspects to find the reason.

**Comment by liminghao1630** [comment]
I think I know why, our best results are based on the second-stage pre-training models and fine-tuned, only the first-stage model is released. Please referring to the "Pre-training" section for more detail.

**Comment by XHMY** [comment]
This is the reason. I read your paper again and found that I overlooked the absence of Stage2 pretrained models. Awesome OCR work. Thank you for your reply.

**Comment by dayyass** [comment]
@XHMY, I'm not sure, but it seems to be `CER: 42.9%`, not `CER: 4.29%`based on your logs
<img width="509" alt="image" src="https://user-images.githubusercontent.com/26326659/226742613-19e6fbae-c826-4e50-81ab-d88b58ee22d9.png">

**Comment by cramraj8** [comment]
@liminghao1630  Is the stage-2 pre-trained models are released ?
